### DISCUSSION TITLE
Few question before I start making

### DISCUSSION THREAD

**Discussion by shashankx86** [slint-ui/slint#4226] [Q&A]
Hi,

I making a concept app drawer (Don't ask why, i am just doing it)

The idea is to spawn a window(a) without frame outside of desktop windows leaving bit of in side of desktop window
when mouse is placed on window(a) (fig 1) it should slide to specified (like center left right)
 
### fig 1
<img src="https://github.com/slint-ui/slint/assets/64682801/51982f9e-fb3d-49ab-962f-6cca8f63652b" width="50%"> 

### fig 2
<img src="https://github.com/slint-ui/slint/assets/64682801/5c32a242-1fed-43ea-9e88-fff6d63ea3b1" width="50%">

before i start making i want to ask few questions

1. frame less window possible?
2. positioning of window(a) at spawn?
3. is it possible for window to detect whether it has mouse over it or not
4. [to know: see (fig 1) when mouse pointer goes over it the windows should come at specified location like center, left, etc (see fig 2)]
the windows in fig 1 comes in like fig 2 with like slide animation, it shouldn't just suddenly appear

**Comment by Vadoola** [comment]
I use some techniques to move a window in my project which is recreating the frame manually. In non-Wayland environments you can use some slint private APIs to set the x and y position (as exposed by win it). Due to the way Wayland is designed, this wasn't working and I had to use a different drag window function (which I don't think would work for you)

I never tried animating the set x and y of the frame like you are asking, so I have no clue if it would work using Slint's animation framework or if you would need to handle the animation from code.

Here are some previous discussions that might help point you in the right direction.

https://github.com/slint-ui/slint/discussions/3859

https://github.com/slint-ui/slint/discussions/4094

https://github.com/slint-ui/slint/discussions/3355

**Reply by shashankx86** [reply]
I am on wayland plasma 
Debian 12

**Reply by shashankx86** [reply]
also my target system is linux

**Reply by Vadoola** [reply]
Unfortunately I just don't know enough about this level of the software stack to be much use.

From what I understand, Wayland is structured to have security a lot more in mind than X11 did, and this prevents some things from being done that were easily done in X11.  As I understand it the application itself placing the Window at specific coordinates is not supported by Wayland (and mentioned here in this Winit thread: https://github.com/rust-windowing/winit/discussions/2920). I'm **assuming** that in Wayland placing the window is supposed to be handled by the compositor and not the application itself, but I just don't know the details.

You will also notice in that above linked winit discussion that setting the position of the window **did** work in Slint using the Qt backend...honestly I don't know why this is the case, or what the reasoning / limitations are, and I never dug too deep.

Sorry I can't be of much more help, but hopefully this at least gives you a place to start digging deeper.

**Comment by Horbin-Magician** [comment]
I have tried to implement some similar functionality and I think it would be very difficult to achieve the above using only Slint.

> 1. frame less window possible?

It's possible by using `no-frame` property. But on windows, I didn't find a way to hide the window icon in the taskbar (this may not be your problem).

> 2. positioning of window(a) at spawn?

As @Vadoola, It's really about the backend (Winit or Qt). These backends don't seem to implement animation functionality.

> is it possible for window to detect whether it has mouse over it or not

Slint doesn't seem to provide a similar event interface. I think it has to do with specific implementations for different systems, it's hard to get cross-platform? Maybe try using a specific system level API.

All in all, it's hard to do this with slint alone. Strictly speaking, Slint is more concerned with rendering the content within the window rather than the position, animations of the window. The latter is more relevant to the backend (Winit or Qt).

(Disclaimer: I am likewise a novice)

**Reply by bjorn** [reply]
> But on windows, I didn't find a way to hide the window icon in the taskbar (this may not be your problem).

This should be possible since #4291, see https://github.com/slint-ui/slint/discussions/4284#discussioncomment-8077128 for an example of using platform-specific API (that's about macOS, but on Windows you can use similar code to call [`WindowBuilderExtWindows.with_skip_taskbar`](https://docs.rs/winit/latest/x86_64-pc-windows-msvc/winit/platform/windows/trait.WindowBuilderExtWindows.html#tymethod.with_skip_taskbar) for example).

> Slint doesn't seem to provide a similar event interface.

It did not as of Slint 1.3.2, but since #4078 the [`TouchArea`](https://slint.dev/releases/1.3.2/docs/slint/src/language/builtins/elements#toucharea) calls `pointer-event` with a new `Move` event kind for hover events.

I can't really help regarding Wayland though. I think there is special API for implementing docks and such as opposed to regular application windows, but I don't know the details or how one would interact with those. You might want to check how similar projects deal with this before trying to figure out how to achieve it with Slint.

**Reply by Horbin-Magician** [reply]
Thank you. I plan to test each of the methods you've suggested once version 1.3.3 (or 1.4.0) is released.
